examples/xlio_ultra_api_ping_pong.c

```diff
@@ -972,9 +832,451 @@ static void process_received_message(xlio_socket_t sock, const char *msg, size_t
     }
 }
 
+/*
+ * ===================================
+ * TX Buffer Pool Management Functions
+ * ===================================
+ *
+ * TX buffer pool is an application logic, but is critical for performance.
+ * An efficient TX buffer pool implementation can be complex and needs to take into
+ * account hardware nuances, such as CPU caches, and application architecture (e.g.
+ * multi-threading).
+ *
+ * Why Multiple TX Buffers Are Needed
+ * ----------------------------------
+ * Zero-copy send operations are asynchronous - a buffer cannot be reused immediately
+ * after the send call returns. Application gives ownership to the XLIO stack until
+ * the transmission completes, which is signaled via the completion callback.
+ *
+ * Although it is allowed to reuse a single buffer for static unchanged data, dynamic
+ * data requires multiple buffers.
+ *
+ * Note, XLIO_SOCKET_SEND_FLAG_INLINE changes behavior of the send operation along with
+ * its requirements and guarantees. Please refer to the XLIO Ultra API documentation
+ * for more details.
+ *
+ * TX Buffer Lifecycle
+ * ------------------
+ * Typically, a buffer goes through the following states in its lifecycle:
+ *
+ * 1. POOL INITIALIZATION:
+ *    - Memory allocation
+ *    - This can be a pool of fixed-sized buffers or dynamically allocated buffers from
+ *      a pre-allocated memory block
+ *
+ * 2. MEMORY REGISTRATION:
+ *    - Memory registration of the underlying memory blocks
+ *    - See memory registration section for more details
+ *
+ * 3. BUFFER ALLOCATION:
+ *    - Application allocates buffers on demand
+ *
+ * 4. MESSAGE PREPARATION:
+ *    - Application constructs a message withing the buffer
+ *
+ * 5. TRANSMISSION:
+ *    - Application sends the message using the send operation
+ *    - At this point, ownership on the buffer is transferred to the XLIO stack
+ *
+ * 6. COMPLETION:
+ *    - After completing all the related operations, XLIO generates a completion event
+ *    - Ownership on the buffer is transferred back to the application
+ *    - XLIO doesn't make access to the buffer after the completion event
```

Comment:
XLIO doesn't make access to the buffer after the completion event -> XLIO no longer accesses the buffer after the completion event